### ISSUE TITLE
ALB is created, but kubectl is not updated

### ISSUE THREAD

**Issue by luvpreetsingh** [kubernetes-sigs/aws-alb-ingress-controller#969]
### Description:
After I run `kubectl apply -f ingress.yml` to create my ingress namely `fleet-ingress`, the ALB is created with the given routing, but when I run,

    $ kubectl get ing
    NAME                     HOSTS   ADDRESS                               PORTS   AGE
    fleet-ingress            *                                             80      8m38s
    api-koa-ingress          *       xx-xx.-xx-xx-xx-xxelb.amazonaws.com   80      61s

The alb address for this ingress is not returned when I get the ingresses. It is returned in other ingress objects. Also, my DNS entry is not created.

### Expected Behavior:
The ALB address should be returned in the output of `kubectl get ing` and the DNS entry should be created

### Actual Behavior:
It is not returning the ALB address. It should create the record set in route53.

### Details:
I am using EKS.
Here is my ingress.yaml,

    ---
    apiVersion: extensions/v1beta1
    kind: Ingress
    metadata:
      name: "fleet-ingress"
      annotations:
        kubernetes.io/ingress.class: alb
        alb.ingress.kubernetes.io/scheme: internet-facing
        alb.ingress.kubernetes.io/subnets: subnet-07a503f30f, subnet-08f16eb1935
        # for creating entry in route 53
        external-dns.alpha.kubernetes.io/hostname: xx.yy.zz
        alb.ingress.kubernetes.io/load-balancer-attributes: access_logs.s3.enabled=true,access_logs.s3.bucket=eks-alb,access_logs.s3.prefix=fleet
      labels:
        app: fleet-app
    spec:
      rules:
        - http:
            paths:
              - path: /static/*
                backend:
                  serviceName: "staticfiles-fleet"
                  servicePort: 50000
              - path: /api/*
                backend:
                  serviceName: "fleet-app-nodeport"
                  servicePort: 49000
              - path: /dashboard/*
                backend:
                  serviceName: "fleet-app-nodeport"
                  servicePort: 49000

The ALB is created with all the routing. **But the DNS entry is not created**. I have used the same ingress file in other ingress objects and it worked fine completely, creating the DNS entry 
also.

**No Errors are shown in the description**. Here is the description of the ingress,

    Name:             fleet-ingress
    Namespace:        default
    Address:
    Default backend:  default-http-backend:80 (<none>)
    Rules:
      Host  Path  Backends
      ----  ----  --------
      *
            /static/*          staticfiles-fleet:50000 (10.0.2.63:80)
            /api/*             fleet-app-nodeport:49000 (10.0.2.25:9000)
            /dashboard/*   fleet-app-nodeport:49000 (10.0.2.25:9000)
    Annotations:
      alb.ingress.kubernetes.io/load-balancer-attributes:  access_logs.s3.enabled=true,access_logs.s3.bucket=eks-alb,access_logs.s3.prefix=fleet-eks
      alb.ingress.kubernetes.io/scheme:                    internet-facing
      alb.ingress.kubernetes.io/subnets:                   subnet-07a503f30f, subnet-08f16eb1935
      kubectl.kubernetes.io/last-applied-configuration:    {"apiVersion":"extensions/v1beta1","kind":"Ingress","metadata":{"annotations":{"alb.ingress.kubernetes.io/load-balancer-attributes":"access_logs.s3.enabled=true,access_logs.s3.bucket=eks-alb,access_logs.s3.prefix=fleet-eks","alb.ingress.kubernetes.io/scheme":"internet-facing","alb.ingress.kubernetes.io/subnets":"subnet-07a503f30f, subnet-08f16eb1935","kubernetes.io/ingress.class":"alb"},"labels":{"app":"fleet-app"},"name":"fleet-ingress","namespace":"default"},"spec":{"rules":[{"http":{"paths":[{"backend":{"serviceName":"staticfiles-fleet","servicePort":50000},"path":"/static/*"},{"backend":{"serviceName":"fleet-app-nodeport","servicePort":49000},"path":"/api/*"},{"backend":{"serviceName":"fleet-app-nodeport","servicePort":49000},"path":"/A92DFFBB6B9EC/*"}]}}]}}

      kubernetes.io/ingress.class:  alb
    Events:
      Type    Reason  Age   From                    Message
      ----    ------  ----  ----                    -------
      Normal  CREATE  4s    alb-ingress-controller  LoadBalancer xxab12xx-default-fleetingr-4c24 created, ARN: arn:aws:elasticloadbalancing:us-west-2:xxx:loadbalancer/app/xxab12xx-default-fleetingr-4c24/aabb1122xx22
      Normal  CREATE  3s    alb-ingress-controller  rule 1 created with conditions [{    Field: "path-pattern",    Values: ["/static/*"]  }]
      Normal  CREATE  3s    alb-ingress-controller  rule 2 created with conditions [{    Field: "path-pattern",    Values: ["/api/*"]  }]
      Normal  CREATE  2s    alb-ingress-controller  rule 3 created with conditions [{    Field: "path-pattern",    Values: ["/A92DFFBB6B9EC/*"]  }]

To check whether my **external-dns** is working or not, I created my older ingress and created it again. It worked perfectly fine. 
The services and pods are all working fine, 100%.

Can you help in debugging this? I'm feeling helpless here as the same config is working for others.

**Comment by luvpreetsingh** [comment]
This worked automatically after some time. I have to say it is inconsistent at times.

**Comment by JuanAnsio** [comment]
Same here. This is very unreliable, so we had to create a shell script to get ALB endpoints. Can someone please take a look at the reason? I guess that ingress resource is not updated with endpoint url for any reason, timeout maybe?

**Comment by M00nF1sh** [comment]
@JuanAnsio There should be some logs in the controller pod. ALB is created before other resources update(like security rules), so there should be error messages to tell what's going wrong

**Comment by ertenmehmet** [comment]
Have you ever managed to work ALB to serve other than 80 and/or 443?? They have explained with such quote "An Ingress does not expose arbitrary ports or protocols. Exposing services other than HTTP and HTTPS to the internet typically uses a service of type NodePort or LoadBalancer." in their EKS training.

**Comment by kevinkarwaski** [comment]
FWIW, the issue I found is that if you define a service in a rule under the ingress spec that does not yet exist, the ingress controller creates the ALB but thrashes on creating target groups for the non-existent service. I noticed this in the controller pod logs. The errors looked like this:

`E1002 20:36:21.666967       1 controller.go:217] kubebuilder/controller "msg"="Reconciler error" "error"="failed to reconcile targetGroups due to failed to load serviceAnnotation due to no object matching key \"default/my-service\" in local store"  "controller"="alb-ingress-controller" "request"={"Namespace":"default","Name":"alb-ingress-controller"}`

The address field in the ingress wouldn't populate with the ALB dns name until these reconciler errors handled. Once the service exists and the ingress controller can successfully create the target group(s), the ALB url then appears in the address field of the ingress.
